optapy-core/src/main/java/org/optaplanner/optapy/MirrorWithExtrasMap.java

```diff
@@ -62,6 +67,10 @@ public Value_ remove(Object o) {
 
     @Override
     public void putAll(Map<? extends Key_, ? extends Value_> map) {
+        if (!Collections.disjoint(delegateMap.keySet(), map.keySet())) {
+            throw new IllegalArgumentException("Cannot put all entries for map (" + map + ") because it shares "
+                    + " some keys with the mirrored map (" + delegateMap + ").");
+        }
         extraEntriesMap.putAll(map);
```

Comment:
I'd consider reimplementing this as `foreach(... -> put(...))`.
You don't have to repeat the fail-fast, and doing the disjoint call is needlessly expensive.

Also, let's agree that - unless there's a good reason for doing it otherwise, class fields will be `private`, OK?

Reply:
We can do the `private` stuff in future PRs, no need to do that now.